packages/data-schema/src/SchemaProcessor.ts

```diff
@@ -904,17 +906,25 @@ function mapToNativeAppSyncAuthDirectives(
     const provider = getAppSyncAuthDirectiveFromRule(rule);
 
     if (rule.groups) {
-      // example: (cognito_groups: ["Bloggers", "Readers"])
+      if(!groupProvider.has(provider)) {
+        groupProvider.set(provider, new Set());
+      };
+      rule.groups.forEach((group) => groupProvider.get(provider)?.add(group));
+    } else {
+      rules.add(provider);
+    }
+
+    groupProvider.forEach((groups, provider) => {
       rules.add(
-        `${provider}(cognito_groups: [${rule.groups
+        `${provider}(cognito_groups: [${Array.from(groups.keys())
           .map((group) => `"${group}"`)
           .join(', ')}])`,
       );
-    } else {
-      rules.add(provider);
-    }
+      // example: (cognito_groups: ["Bloggers", "Readers"])
+    })
```

Comment:
Small gripes from me here that the `forEach` on a `Map` and the `.keys()` on the `Set` feel like the less intuitive options here. But, not necessarily blocking.

Reply:
Turns out I can drop the keys without changing the behavior. I have done so.